<?xml version="1.0" ?>
<robot name="palletizer_model" xmlns:xacro="http://www.ros.org/wiki/xacro">
	<link name ="world"/>
	<joint name="pal_base" type="fixed">
		<parent link="world"/>
		<child link="pal_link0"/>
		<origin xyz = "0 300 0" rpy="0 0 -1.57" />
		<axis xyz="0 1 0" />
	</joint>

	<link name="pal_link0">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link0.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link0.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint0" type="continuous">
		<parent link="pal_link0"/>
		<child link="pal_link1"/>
		<origin xyz = "0 0 107" rpy="0 0 1.57" />
		<axis xyz="0 0 1" />
	</joint>
	<link name="pal_link1">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link1.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link1.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint1" type="revolute">
		<parent link="pal_link1"/>
		<child link="pal_link2"/>
		<origin xyz = "-6.5 -21.131 46.62" rpy="0 -1.57 0" />
		<axis xyz="0 0 -1" />
		<limit effort="1.0" lower="-1.57" upper="1.57" velocity="5.0"/>
	</joint>
	<link name="pal_link2">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link2.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link2.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint2" type="revolute">
		<parent link="pal_link1"/>
		<child link="pal_link8"/>
		<origin xyz = "6.5 -21.13 46.36" rpy="0 0 0" />
		<axis xyz="-1 0 0" />
		<limit effort="1.0" lower="-1.57" upper="1.57" velocity="5.0"/>
	</joint>
	<link name="pal_link3">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link3.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link3.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<link name="pal_link4">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link4.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link4.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint3" type="revolute">
		<parent link="pal_link4"/>
		<child link="pal_link4_1"/>
		<origin xyz = "0 -49.5 -17" rpy="0 0 0" />
		<axis xyz="0 0 -1" />
		<limit effort="1.0" lower="-3.14" upper="3.14" velocity="5.0"/>
	</joint>
	<link name="pal_link4_1">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link4_1.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link4_1.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="prisos_joint" type="revolute">
		<parent link="pal_link4_1"/>
		<child link="prisos"/>
		<origin xyz = "1 -41 -66" rpy="0 0 -1.57" />
		<axis xyz="0 0 -1" />
		<limit effort="1.0" lower="-3.14" upper="3.14" velocity="5.0"/>
	</joint>
	<link name="prisos">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/prisos.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/prisos.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint4" type="continuous">
		<parent link="pal_link3"/>
		<child link="pal_link4"/>
		<origin xyz = "-160 0 0" rpy="-1.57 -1.57 0" />
		<axis xyz="1 0 0" />
		<mimic joint="pal_joint1" multiplier="-1" offset="0"/>
	</joint>
	<joint name="pal_joint10" type="continuous">
		<parent link="pal_link1"/>
		<child link="pal_link5"/>
		<origin xyz = "33.9 13.28 70.418" rpy="1.57 0 3.14" />
		<axis xyz="1 0 0" />
		<mimic joint="pal_joint1" multiplier="1" offset="0"/>
	</joint>
	<link name="pal_link5">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link5.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link5.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint5" type="continuous">
		<parent link="pal_link2"/>
		<child link="pal_link6"/>
		<origin xyz = "148 0 -20.5" rpy="0 0 0" />
		<axis xyz="0 0 1" />
		<mimic joint="pal_joint3" multiplier="-1" offset="0"/>
	</joint>
	<link name="pal_link6">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link6.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link6.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint6" type="continuous">
		<parent link="pal_link4"/>
		<child link="pal_link7"/>
		<origin xyz = "24 -34.404 24.09" rpy="3.14 0 0" />
		<axis xyz="1 0 0" />
		<mimic joint="pal_joint3" multiplier="-1" offset="0"/>
	</joint>
	<link name="pal_link7">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link7.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link7.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint7" type="continuous">
		<parent link="pal_link2"/>
		<child link="pal_link3"/>
		<origin xyz = "148 0 0" rpy="0 3.14 -1.57" />
		<axis xyz="0 0 1" />
		<mimic joint="pal_joint2" multiplier="1" offset="0"/>
	</joint>

	<link name="pal_link8">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link8.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link8.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>
	<joint name="pal_joint8" type="continuous">
		<parent link="pal_link8"/>
		<child link="pal_link9"/>
		<origin xyz = "-21 54 0" rpy="1.57 0 0" />
		<axis xyz="1 0 0" />
		<mimic joint="pal_joint1" multiplier="-1" offset="0"/>
	</joint>
	<link name="pal_link9">
		<collision>
			<origin xyz = "0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link9.stl" scale="1 1 1"/>
			</geometry>
		</collision>
		<visual>
			<geometry>
				<mesh filename="package://palletizer_arm_model/meshes/palletizer/link9.stl" scale="1 1 1"/>
			</geometry>
			<material name="grey">
    				<color rgba="0.7 0.7 0.7 1.0"/>
  			</material>
		</visual>
	</link>

</robot>
